lib/helpers/common.js

```diff
@@ -67,7 +69,22 @@ function flatten(update, path, options, schema) {
  * ignore
  */
 
-function modifiedPaths(update, path, result) {
+function modifiedPaths(update, path, result, recursion = null) {
+  if (recursion == null) {
+    recursion = {
+      raw: { update, path },
+      trace: new WeakSet()
+    };
+  }
+
+  if (recursion.trace.has(update)) {
+    throw new MongooseError(`a circular reference in the update value, updateValue:
+${util.inspect(recursion.raw.update, { showHidden: false, depth: 1 })}
+updatePath: ${recursion.raw.path}
+-------`);
+  }
+  recursion.trace.add(update);
```

Comment:
Add a check for `if (update == null) { return }` before this. Adding `null` or `undefined` to a `WeakSet()` throws an error, so we should avoid this case.

Reply:
Sorry, I don't know this. I added a null check.